core/src/main/java/io/substrait/relation/Extension.java

```diff
@@ -0,0 +1,57 @@
+package io.substrait.relation;
+
+import io.substrait.type.NamedStruct;
+import io.substrait.type.Type;
+import java.util.List;
+
+/** Contains tag interfaces for handling {@link com.google.protobuf.Any} types within Substrait. */
+public class Extension {
+
+  private interface ToProto {
```

Comment:
I'd move this out of Extension and probably apply it consistently throughout the codebase. (the move out could be done here, application elsewhere as a separate piece of work).

Reply:
I suggest that we modify the interface to ToProto<T extends Message>. We have the toProto pattern in a lot of locations throughout the java codebase.

Reply:
Gotcha. The `T extends Message` was what I was missing. It wasn't immediately clear to me how the existing `ToProto` interface could work for serializing to different message types. Parametrizing it with a generic makes sense. I've made the change.